components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4237,6 +4237,29 @@ public static String getAccessTokenIdentifier(OAuth2TokenValidationResponseDTO t
         return null;
     }
 
+    /**
+     * Return access token DO from OAuth2TokenValidationResponseDTO. This method validated the token against DB or
+     * in-memory.
+     *
+     * @param tokenResponse OAuth2TokenValidationResponseDTO object.
+     * @return extracted access token DO.
+     * @throws UserInfoEndpointException if  error occurred while obtaining access token.
+     */
+    public static AccessTokenDO getAccessTokenDO(OAuth2TokenValidationResponseDTO tokenResponse)
+            throws UserInfoEndpointException {
+
+        AccessTokenDO accessTokenDO = null;
+        if (tokenResponse.getAuthorizationContextToken().getTokenString() != null) {
```

Comment:
Is there is any chance for  tokenResponse or tokenResponse.getAuthorizationContextToken() to be null add a null check to that